tests/Unit/Time/Test_TakeStep.cpp

```diff
@@ -0,0 +1,127 @@
+  // prepare history so that the Adams-Bashforth is ready to take steps
+  TimeStepperTestUtils::initialize_history(
+      slab.start(), make_not_null(&history),
+      [&initial_values](const auto t) noexcept {
+        return initial_values * exp(t);
+      },
+      [](const auto y, const auto /*t*/) noexcept { return y; }, time_step, 4);
+
+  auto box = db::create<db::AddSimpleTags<
+      Tags::TimeStepId, Tags::Next<Tags::TimeStepId>, Tags::TimeStep,
+      Tags::Next<Tags::TimeStep>, EvolvedVariable, Tags::dt<EvolvedVariable>,
+      Tags::HistoryEvolvedVariables<EvolvedVariable>,
+      Tags::TimeStepper<LtsTimeStepper>, Tags::StepChoosers<step_chooser_list>,
+      Tags::StepController>>(
+      TimeStepId{true, 0_st, slab.start()},
+      TimeStepId{true, 0_st, Time{slab, {1, 2}}}, TimeDelta{slab, {1, 2}},
+      TimeDelta{slab, {1, 2}}, initial_values, DataVector{5, 0.0},
+      std::move(history),
+      static_cast<std::unique_ptr<LtsTimeStepper>>(
+          std::make_unique<TimeSteppers::AdamsBashforthN>(5)),
+      std::move(step_choosers),
+      static_cast<std::unique_ptr<StepController>>(
+          std::make_unique<StepControllers::BinaryFraction>()));
+
+  // update the rhs
+  db::mutate<Tags::dt<EvolvedVariable>>(make_not_null(&box), update_rhs,
+                                         db::get<EvolvedVariable>(box));
+  take_step<step_chooser_list>(make_not_null(&box), cache);
+  // check that the state is as expected
+  CHECK(db::get<Tags::Next<Tags::TimeStep>>(box) == TimeDelta{slab, {1, 4}});
+  CHECK(db::get<EvolvedVariable>(box) == initial_values * exp(0.005));
+  CHECK(db::get<Tags::dt<EvolvedVariable>>(box) == initial_values);
+
+  // advance time
+  db::mutate<Tags::TimeStepId, Tags::Next<Tags::TimeStepId>, Tags::TimeStep,
+             Tags::Next<Tags::TimeStep>>(
+      make_not_null(&box),
+      [](const gsl::not_null<TimeStepId*> time_id,
+         const gsl::not_null<TimeStepId*> next_time_id,
+         const gsl::not_null<TimeDelta*> local_time_step,
+         const gsl::not_null<TimeDelta*> next_time_step,
+         const TimeStepper& time_stepper) noexcept {
+        *time_id = *next_time_id;
+        *local_time_step =
+            next_time_step->with_slab(time_id->step_time().slab());
+
+        *next_time_id =
+            time_stepper.next_time_id(*next_time_id, *local_time_step);
+        *next_time_step =
+            local_time_step->with_slab(next_time_id->step_time().slab());
+      },
+      db::get<Tags::TimeStepper<>>(box));
+
+  db::mutate<Tags::dt<EvolvedVariable>>(make_not_null(&box), update_rhs,
+                                        db::get<EvolvedVariable>(box));
+  take_step<step_chooser_list>(make_not_null(&box), cache);
+  // check that the state is as expected
+  CHECK(db::get<Tags::TimeStepId>(box).substep_time().value() == 0.005);
+  CHECK(db::get<Tags::Next<Tags::TimeStep>>(box) == TimeDelta{slab, {1, 8}});
```

Comment:
I don't understand why this is the outcome.  We've just taken a 1/4 step from 0.005 to 0.0075, and I think this line means we will next take a 1/8 step, which will put us at 0.00875.  But one of the goals was 0.00751, which we've then missed.